### DISCUSSION TITLE
Linux: Missing dependencies libgeotiff.so.5 when executing AppImage 3.9.5

### DISCUSSION THREAD

**Discussion by Area5142** [sn4k3/UVtools#637] [Q&A] [answered]
### System

```Text
UVtools v3.9.5 X64
Operative system: Linux 5.15.0-56-generic #62-Ubuntu SMP Tue Nov 22 19:54:14 UT>
Processor: Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz
Processor cores: 4
Memory RAM: 7.62 / 7.73 GB
Runtime: linuxmint.21.1-x64
Framework: .NET 6.0.12
AvaloniaUI: 0.10.18
OpenCV: 4.6.0

Sreens, resolution, working area, usable area:
1: 1600 x 1200 @ 100% (Primary)
    WA: 1600 x 1160    UA: 1600 x 1160


Path:       /tmp/.mount_UVtoolocEiY9/usr/bin/
Executable: /tmp/.mount_UVtoolocEiY9/usr/bin/UVtools
```

### Printer and Slicer

- Printer: Prusa SL1S Speed
- Slicer: PrusaSlicer 2.5.0

### Description of the bug

I get the following error when executing the AppImage (3.9.5 x64) before entering the application and main screen:
![image](https://user-images.githubusercontent.com/1320126/210287682-f67066b5-c23f-41c1-ac6d-95fb5858745e.png)

The same happens with version: 3.9.4 - 3.9.0, 3.8.3 - 3.8.0 (have not tried earlier versions.

### How to reproduce

1. Install Linux Mint 21.1 x64 version on PC or Image.
2. Download UVtools_linux-x64_v3.9.5.AppImage and make executable.
3. Run AppImage and the error screen appears.

As I read the error message, some files is missing in the AppImage (libgeotiff.so.5).

**Comment by sn4k3** [comment, marked as answer]
You have to install the required dependencies, either by run the [auto install script](https://github.com/sn4k3/UVtools#to-auto-install-on-linux) or [dependency install script](https://github.com/sn4k3/UVtools#linux) or manually.

**Reply by Area5142** [reply]
Was aware that there was a reference to the help guide about pre-installing software - didn't think that applied to AppImages, which usually has everything included.

UVTools 3.10.0 starts up if I install:

`sudo apt install libgeotiff-dev`

I'll find out if that's enough in the next few days.

It would be nice if there was a list of dependencies so you could manually install the missing packages without having to try to read a shell script (and probably fail ;).

Thanks for the great work on UVTools, I'm really looking forward to getting started with it.

**Reply by sn4k3** [reply]
> didn't think that applied to AppImages, which usually has everything included.

AppImage is just a zip that extract to a folder and run, the requirements are that the AppImage run in any fresh system without any dependency requirement, and then UVtools fails on that but it's added for user conveniency.

> It would be nice if there was a list of dependencies so you could manually install the missing packages without having to try to read a shell script (and probably fail ;).

That's why there is a shell script, for people run and have that problem fixed, you just need to trust it ;)
The list of missing dependencies are shown by UVtools, you just need to grab it name without extension, if not work append `-dev`

**Reply by Area5142** [reply]
One of the reasons for my Linux Mint upgrade from 19.3 to 21.1 is to be able to run UVTools.
Previously I ran UVTools in a VM, but that's too cumbersome...

Is it correctly understood that all dependencies are in place when UVTools starts without complaining?
Or is it something that comes along the way when the various functions are put into use?

Thanks for your quick reply ;-)

**Reply by sn4k3** [reply]
A major OS upgrade is always welcome 😄

> Is it correctly understood that all dependencies are in place when UVTools starts without complaining?

Yes, only one tool won't work. Layer images to animated gif. But I think no one uses that...

**Reply by Area5142** [reply]
Thank you for the help. Have a nice day :)

**Reply by leafish-dylan** [reply]
libgeotiff pulls in 600MB of further dependencies in Fedora, and presumably other distros. This is not great for Silverblue or similar immutable Linux versions, where the intention is to keep the base OS minimal and use containers (eg, flatpak) for applications. I can build a flatpak version of UVTools, or use a Toolbox container, but it's big.

I'm not sure why UVTools needs an image geolocation library, but it would be great if this could be looked at to see if it's necessary.

**Reply by sn4k3** [reply]
libgeotiff by default is required due opencv library, but I plan to remove it when upgrading OpenCV version
